src/lib/net/SecureSocket.cpp

```diff
@@ -610,70 +610,50 @@ void SecureSocket::disconnect()
 {
-  // calculate received certificate fingerprint
-  using AutoX509 = std::unique_ptr<X509, decltype(&X509_free)>;
-  AutoX509 cert(SSL_get_peer_certificate(m_ssl->m_ssl), &X509_free);
-
-  unsigned char tempFingerprint[EVP_MAX_MD_SIZE];
-  unsigned int tempFingerprintLen;
-  int digestResult = X509_digest(cert.get(), EVP_sha256(), tempFingerprint, &tempFingerprintLen);
-
-  if (digestResult <= 0) {
-    LOG((CLOG_ERR "failed to calculate fingerprint, digest result: %d", digestResult));
+  deskflow::FingerprintData sha1;
+  deskflow::FingerprintData sha256;
+  try {
+    auto cert = SSL_get_peer_certificate(m_ssl->m_ssl);
+    sha1 = deskflow::sslCertFingerprint(cert, deskflow::FingerprintType::SHA1);
+    sha256 = deskflow::sslCertFingerprint(cert, deskflow::FingerprintType::SHA256);
+  } catch (const std::exception &e) {
+    LOG((CLOG_ERR "%s", e.what()));
     return false;
   }
 
-  // format fingerprint into hexdecimal format with colon separator
-  std::string fingerprint(static_cast<char *>(static_cast<void *>(tempFingerprint)), tempFingerprintLen);
-  formatFingerprint(fingerprint);
-  LOG((CLOG_NOTE "server fingerprint: %s", fingerprint.c_str()));
+  // Gui Must Parse these two lines, DO NOT CHANGE
+  LOG(
+      (CLOG_NOTE "server fingerprint: (SHA1) %s (SHA256) %s", deskflow::formatSSLFingerprint(sha1.data).c_str(),
+       deskflow::formatSSLFingerprint(sha256.data).c_str())
+  );
 
-  std::string trustedServersFilename;
-  trustedServersFilename = deskflow::string::sprintf(
-      "%s/%s/%s", ARCH->getProfileDirectory().c_str(), kFingerprintDirName, kFingerprintTrustedServersFilename
+  std::string trustedServersFilename = deskflow::string::sprintf(
+      "%s/%s/%s", ARCH->getProfileDirectory().c_str(), kSslDir, kFingerprintTrustedServersFilename
   );
 
   // check if this fingerprint exist
   std::string fileLine;
   std::ifstream file;
-  file.open(deskflow::filesystem::path(trustedServersFilename));
-
-  bool isValid = false;
-  if (file.is_open()) {
-    while (!file.eof()) {
-      getline(file, fileLine);
-      if (!fileLine.empty() && !fileLine.compare(fingerprint)) {
-        isValid = true;
-        break;
-      }
-    }
+
+  deskflow::openUtf8Path(file, trustedServersFilename);
+  deskflow::FingerprintDatabase db;
+  db.read(trustedServersFilename);
+  if (!db.fingerprints().empty()) {
+    LOG((CLOG_NOTE "read %d fingerprints from %s", db.fingerprints().size(), trustedServersFilename.c_str()));
   } else {
-    LOG((CLOG_ERR "fail to open trusted fingerprints file: %s", trustedServersFilename.c_str()));
+    LOG((CLOG_ERR "failed to open trusted fingerprints file: %s", trustedServersFilename.c_str()));
+    return false;
   }
 
-  file.close();
-  return isValid;
+  if (!db.isTrusted(sha256) || !db.isTrusted(sha1)) {
+    LOG((CLOG_WARN "fingerprint does not match trusted fingerprint"));
+    return false;
+  }
+
+  LOG((CLOG_NOTE "fingerprint matches trusted fingerprint"));
+  return true;
```

Comment:
Alternative idea: thos might be more readable, and therefore maintainable, and therefore more secure. 

It also allows us to log a warning when a weak SHA1 fingerprint is used.

```suggestion
  if (db.isTrusted(sha256)) {
    LOG((CLOG_NOTE "fingerprint matches SHA256 trusted fingerprint"));
    return true;
  }

  if (db.isTrusted(sha1)) {
    LOG((CLOG_WARN "fingerprint matches weak SHA1 trusted fingerprint, use SHA256 instead"));
    return true;
  }

  LOG((CLOG_WARN "fingerprint does not match trusted fingerprint"));
  return false;
```